pkg/checkpoint/projected.go

```diff
@@ -0,0 +1,77 @@
+package checkpoint
+
+import (
+	"fmt"
+
+	"k8s.io/api/core/v1"
+)
+
+func (c *checkpointer) checkpointProjectedVolumes(pod *v1.Pod) (*v1.Pod, error) {
+	uid, gid, err := podUserAndGroup(pod)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get user and group for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+	}
+
+	for i := range pod.Spec.Volumes {
+		v := &pod.Spec.Volumes[i]
+
+		if v.VolumeSource.Projected != nil {
+			p := v.VolumeSource.Projected
+			if volumeHasSecretProjection(v) {
+				err := c.checkpointProjectedSecrets(pod.Namespace, pod.Name, v.Name, p.Sources, uid, gid)
+				if err != nil {
+					return nil, fmt.Errorf("failed to checkpoint projected secrets for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+				}
+			}
+			if volumeHasConfigMapProjection(v) {
+				err := c.checkpointProjectedConfigMaps(pod.Namespace, pod.Name, v.Name, p.Sources, uid, gid)
+				if err != nil {
+					return nil, fmt.Errorf("failed to checkpoint projected configmaps for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+				}
+			}
+			if volumeHasDownwardAPIProjection(v) {
+				return nil, fmt.Errorf("no support to checkpoint projected DownwardAPI for pod %s/%s", pod.Namespace, pod.Name)
```

Comment:
Not supporting all types was brought up as a concern. Are we okay with shipping without support for this type? At least some support is better than none, but want to make sure other's okay with this.

cc @dghubble

Reply:
The not supporting downward API projection part sounds fine at least, since it wasn't supported before. If folks using kubeadm or other custom tools use manifests with downwardapi projections, they could contemplate how to implement it.